src/illumidesk/authenticators/authenticator.py

```diff
@@ -172,8 +175,7 @@ async def authenticate(self, handler: BaseHandler, data: Dict[str, str] = None)
             # runs as a docker container we need to normalize the string so we can use it
             # as a container name.
             if 'context_label' in args and args['context_label'] is not None:
-                course_label = args['context_label']
-                course_id = lti_utils.normalize_name_for_containers(course_label)
+                course_id = args['context_label']
                 self.log.debug('Course context_label normalized to: %s' % course_id)
```

Comment:
ok, I understand that the use of the `setup_course_hook` is required in the existing Authenticator classes to use, but what will happen if a new class of authenticator never calls that hook?  the names could be wrong right? Why not return the **course**, **username** always in lower case (or normalized) from our authenticators and checking it too in the hook? 
what do you think?

Reply:
We could, but the root cause for the bug in the first place was because we were calling the normalize_string function point-to-point from the different authenticators and missed one of the calls for the course label. So as we add more items (potentially) to the `auth_state` dict then it gets harder to maintain.

I think for our setup we can assume that all authenticators will inherit from our base spawner class.

Reply:
but i remember that the auth_state dict is used by other classes and flows. I'm not sure if the `name` with spaces or in upper case will affected other parts.. but in general this change seems good